integration/v3_barrier_test.go

```diff
@@ -47,30 +47,38 @@ func testBarrier(t *testing.T, waiters int, chooseClient func() *clientv3.Client
 	}
 
 	donec := make(chan struct{})
+	stopc := make(chan struct{})
```

Comment:
add a defer to close the stopc instead of calling it before each fatal?

fatalnow will run the defered func: see https://golang.org/pkg/testing/#T.FailNow

Reply:
Good idea! Thanks for the suggestion. I have already modified the patch.